lib/apiservers/portlayer/restapi/handlers/interaction_handlers.go

```diff
@@ -238,8 +291,9 @@ func (d *FlushingReader) AddFlusher(flusher GenericFlusher) {
 func (d *FlushingReader) WriteTo(w io.Writer) (written int64, err error) {
 	buf := make([]byte, 64)
 
+	nr, er := d.readDetectInit(buf)
```

Comment:
It's permitted for Read/WriteTo/ReadFrom to return `0, nil` - I'm not aware of when that will occur with stdlib functions.
Placing this here means we will attempt to filter initial bytes for every call to WriteTo, instead of the initial bytes for the Reader as a whole - is this intentional?